---
layout: post
title: "Python functions for microbiome exploratory data analysis."
date: 2017-04-17
use_math: true
---

Over the past couple of years, I have been repeatedly munging through microbiome data and also repeatedly re-writing the same code to analyze similar datasets. On more occasions, I have been asked by peers and colleagues to share bits of the code that I have written. Eventually, I forced mysefl to create a central repository for code routinely used by me to analyze 16S amplicon sequencing data sets. In the David lab, we run our sequencing results through the [QIIME](http://qiime.org) pipeleine and accordingly most of the data analysis begins with a [BIOM](http://biom-format.org) file.

Here, I am sharing some of the primary functions that make exploration and analysis of microbiome data simpler. In particular,  

Here is a tutorial of these functions in a [jupyter notebook](/assets/ipynbs/2017_04_10_seq_analysis_post.html). 
